phoenix-core/src/main/java/org/apache/phoenix/coprocessor/IndexRebuildRegionScanner.java

```diff
@@ -357,6 +361,7 @@ public boolean next(List<Cell> results) throws IOException {
         } catch (Throwable e) {
             LOGGER.error("Exception in IndexRebuildRegionScanner for region "
                     + region.getRegionInfo().getRegionNameAsString(), e);
+            this.shouldRetry = true;
```

Comment:
Are we sure this is the only path where we have to set this variable ? Are there no other exception paths ? In general it is difficult to catch all exception paths. An alternative could be to just handle the successful case and retry all other times.

Reply:
Actually this is the only path where we set this variable. Any exception in verify or rebuild or region split or move in this loop will set it. I am not sure how we can detect just the successful case since we don't know the success case as you said. What do we consider as success case? So I think leaving it like this will cover most of the exception cases.

Reply:
I think instead of the exception approach and defining a new field shouldRetry,  we can use the existing hasMore field and pass hasMore instead shouldRetry to logToIndexToolResultTable().